src/Engine/ProtoCore/Namespace/SymbolTable.cs

```diff
@@ -103,6 +242,28 @@ public override bool Equals(object obj)
         }
 
         /// <summary>
+        /// Equality operator overload to have same equality behaviour as object.Equals() override
+        /// </summary>
+        /// <param name="thisSymbol"></param>
+        /// <param name="otherSymbol"></param>
+        /// <returns></returns>
+        public static bool operator ==(Symbol thisSymbol, Symbol otherSymbol)
```

Comment:
Well that's a C# thing (best practice) to overload the equality operator if you're overriding `object.Equals` so that both methods have the same behavior for the given type so as not to confuse users of that type.